### PR TITLE
csv: introduce kubevirtVirtLauncherOSVersion flag

### PR DESCRIPTION
**Pull request by dasionov**
**What this PR does / why we need it**:
This PR introduces a new flag called kubevirtVirtLauncherOSVersion to be used to initialize new environment Variable that will hold the value of the VIRT_LAUNCHER_OS_VERSION and it will be fed from the csv

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:
```jira-ticket
https://issues.redhat.com/browse/CNV-15792
```

**Release note**:

```release-note
None
```

### CODE REVIEW

File: tools/csv-merger/csv-merger.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ var (
 	crdDir                        = flag.String("crds-dir", "", "the directory containing the CRDs for apigroup validation. The validation will be performed if and only if the value is non-empty.")
 	hcoKvIoVersion                = flag.String("hco-kv-io-version", "", "KubeVirt version")
 	kubevirtVersion               = flag.String("kubevirt-version", "", "Kubevirt operator version")
+	kubevirtVirtLauncherOSVersion = flag.String("kubevirt-virt-launcher-os-version", "", "Kubevirt Virt launcher OS version")
```
Comment (nunnatsa) [inline tools/csv-merger/csv-merger.go:121]: nit: I think maybe the parameter and the variable names are too long. WDYT about droping `kubevirt` from the name, or replace it with `kv`?

File: tools/manifest-templator/manifest-templator.go [hunk 2]
```diff
@@ -50,34 +50,35 @@ var (
...
+	webhookImage                  = flag.String("webhook-image", "", "HyperConverged Cluster Webhook image")
+	cliDownloadsImage             = flag.String("cli-downloads-image", "", "Downloads Server image")
+	kvVirtIOWinImage              = flag.String("kv-virtiowin-image-name", "", "KubeVirt VirtIO Win image")
+	smbios                        = flag.String("smbios", "", "Custom SMBIOS string for KubeVirt ConfigMap")
+	machinetype                   = flag.String("machinetype", "", "Custom MACHINETYPE string for KubeVirt ConfigMap (Deprecated, use amd64-machinetype)")
+	amd64MachineType              = flag.String("amd64-machinetype", "", "Custom AMD64_MACHINETYPE string for KubeVirt ConfigMap")
+	arm64MachineType              = flag.String("arm64-machinetype", "", "Custom ARM64_MACHINETYPE string for KubeVirt ConfigMap")
+	hcoKvIoVersion                = flag.String("hco-kv-io-version", "", "KubeVirt version")
+	kubevirtVersion               = flag.String("kubevirt-version", "", "Kubevirt operator version")
+	kubevirtVirtLauncherOSVersion = flag.String("kubevirt-virt-launcher-os-version", "", "Kubevirt Virt launcher OS version")
```
Comment (nunnatsa) [inline tools/manifest-templator/manifest-templator.go:73]: same

File: tools/csv-merger/csv-merger.go [hunk 3]
```diff
@@ -514,27 +515,28 @@ func getCsvBaseParams(replaces string, version semver.Version) *components.CSVBa
...
+		KVUIProxyImage:               *kvUIProxyImage,
+		ImagePullPolicy:              "IfNotPresent",
+		VirtIOWinContainer:           *kvVirtIOWinImage,
+		Smbios:                       *smbios,
+		Machinetype:                  *machinetype,
+		Amd64MachineType:             *amd64MachineType,
+		Arm64MachineType:             *arm64MachineType,
+		HcoKvIoVersion:               *hcoKvIoVersion,
+		KubevirtVersion:              *kubevirtVersion,
+		KubevirtVirtLancherOsVersion: *kubevirtVirtLauncherOSVersion,
```
Comment (nunnatsa) [inline tools/csv-merger/csv-merger.go:532]: same

File: pkg/components/components.go [hunk 4]
```diff
@@ -62,30 +62,31 @@ var deploymentType = metav1.TypeMeta{
...
+	ConversionContainer          string
+	VmwareContainer              string
+	VirtIOWinContainer           string
+	Smbios                       string
+	Machinetype                  string
+	Amd64MachineType             string
+	Arm64MachineType             string
+	HcoKvIoVersion               string
+	KubevirtVersion              string
+	KubevirtVirtLancherOsVersion string
```
Comment (nunnatsa) [inline pkg/components/components.go:81]: same

**Review by nunnatsa — COMMENTED** [review]
Thanks for the PR @dasionov 

I added one nit comment (repeatedly). But the main issue with the current version of the PR is that there is no use in the new parameter. The code does nothing with it.

File: tools/csv-merger/csv-merger.go [hunk 5]
```diff
@@ -118,6 +118,7 @@ var (
 	crdDir                        = flag.String("crds-dir", "", "the directory containing the CRDs for apigroup validation. The validation will be performed if and only if the value is non-empty.")
 	hcoKvIoVersion                = flag.String("hco-kv-io-version", "", "KubeVirt version")
 	kubevirtVersion               = flag.String("kubevirt-version", "", "Kubevirt operator version")
+	kvVirtLauncherOSVersion       = flag.String("kubevirt-virt-launcher-os-version", "", "Kubevirt Virt launcher OS version")
```
Comment (nunnatsa) [inline tools/csv-merger/csv-merger.go:121]: Thaks for the change, but the command line parameter name is still too long. please drop `kubevirt-`

File: tools/manifest-templator/manifest-templator.go [hunk 6]
```diff
@@ -50,34 +50,35 @@ var (
...
+	webhookImage            = flag.String("webhook-image", "", "HyperConverged Cluster Webhook image")
+	cliDownloadsImage       = flag.String("cli-downloads-image", "", "Downloads Server image")
+	kvVirtIOWinImage        = flag.String("kv-virtiowin-image-name", "", "KubeVirt VirtIO Win image")
+	smbios                  = flag.String("smbios", "", "Custom SMBIOS string for KubeVirt ConfigMap")
+	machinetype             = flag.String("machinetype", "", "Custom MACHINETYPE string for KubeVirt ConfigMap (Deprecated, use amd64-machinetype)")
+	amd64MachineType        = flag.String("amd64-machinetype", "", "Custom AMD64_MACHINETYPE string for KubeVirt ConfigMap")
+	arm64MachineType        = flag.String("arm64-machinetype", "", "Custom ARM64_MACHINETYPE string for KubeVirt ConfigMap")
+	hcoKvIoVersion          = flag.String("hco-kv-io-version", "", "KubeVirt version")
+	kubevirtVersion         = flag.String("kubevirt-version", "", "Kubevirt operator version")
+	kvVirtLauncherOSVersion = flag.String("kubevirt-virt-launcher-os-version", "", "Kubevirt Virt launcher OS version")
```
Comment (nunnatsa) [inline tools/manifest-templator/manifest-templator.go:73]: same

File: tools/csv-merger/csv-merger.go (hunk 1 above)
Comment (dasionov) [reply tools/csv-merger/csv-merger.go]: done

**Comment by nunnatsa** [comment]
Again, the current version of the code does nothing with the new parameter. The right place to place the new code is here: https://github.com/kubevirt/hyperconverged-cluster-operator/blob/e3b2b427cd5577f09004a50fdb074c5a07005b17/pkg/components/components.go#L186-L280

I would take the `Env` field definition out of the struct initialization; i.e. add a new `envs` slice variable above the `return` in line 160. Then I would add some code to check if the new parameter is not empty, and if it is not, I would append a new env var to the `envs` slice. Feel free to move all of this to a new function. WDYT?

**Comment by dasionov** [comment]
> hyperconverged-cluster-operator/pkg/components/components.go

wow, no idea how I missed that, sorry !

File: pkg/components/components.go [hunk 7]
```diff
@@ -179,105 +182,10 @@ func GetDeploymentSpecOperator(params *DeploymentOperatorParams) appsv1.Deployme
...
-							},
-							{
-								Name:  util.KVUIProxyImageEnvV,
-								Value: params.KVUIProxyImage,
-							},
-							{
-								Name:  util.PrimaryUDNImageEnvV,
-								Value: params.PrimaryUDNImage,
-							},
-						}, params.Env...),
```
Comment (nunnatsa) [inline pkg/components/components.go:280]: Please preserve this logic in the new function: adding the `Env` param values to the end of the slice.

**Review by nunnatsa — APPROVED** [review]
/approve

**Pull request merged**